instrumentation/jvm-metrics/src/main/java/com/splunk/opentelemetry/instrumentation/jvmmetrics/GcMemoryMetrics.java

```diff
@@ -33,33 +29,29 @@
 import java.util.Set;
 import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
 import javax.management.ListenerNotFoundException;
 import javax.management.Notification;
 import javax.management.NotificationEmitter;
 import javax.management.NotificationListener;
 import javax.management.openmbean.CompositeData;
 
-public class GcMemoryMetrics implements MeterBinder, AutoCloseable {
+public class GcMemoryMetrics implements AutoCloseable {
   public static final String METRIC_NAME = "process.runtime.jvm.memory.reclaimed";
   private final boolean managementExtensionsPresent = isManagementExtensionsPresent();
 
   private final List<Runnable> notificationListenerCleanUpRunnables = new CopyOnWriteArrayList<>();
   private final Set<String> heapPoolNames = new HashSet<>();
   private final AtomicLong deltaSum = new AtomicLong();
 
-  @Override
-  public void bindTo(MeterRegistry registry) {
+  public void install(Consumer<AtomicLong> meterFactory) {
     if (!this.managementExtensionsPresent) {
       return;
     }
 
-    GcMetricsNotificationListener gcNotificationListener =
-        new GcMetricsNotificationListener(registry);
+    GcMetricsNotificationListener gcNotificationListener = new GcMetricsNotificationListener();
 
-    FunctionCounter.builder(METRIC_NAME, deltaSum, AtomicLong::get)
-        .description("Sum of heap size differences before and after gc")
-        .baseUnit(BaseUnits.BYTES)
-        .register(registry);
+    meterFactory.accept(deltaSum);
```

Comment:
Same comment as `AllocatedMemoryMetrics` -- seems overly callbacky. Let's just put a method on here to get the value.